python/cudf/cudf/core/column/datetime.py

```diff
@@ -182,6 +183,27 @@ def default_na_value(self):
         """
         return np.datetime64("nat", self.time_unit)
 
+    def mean(self, dtype=np.float64):
+        return pd.Timestamp(
+            self.as_numerical.mean(dtype=dtype), unit=self.time_unit
+        )
+
+    def median(self, dtype=np.float64):
```

Comment:
Could this be defined at the `Column` level? This way all subclasses will raise the same error if they don't implement `median`. Similarly for other reductions.

Reply:
Yea, that change is coming in with this PR: https://github.com/rapidsai/cudf/pull/6137
Once #6137 merges, I can remove this method in this PR